authorizations/message-filter/src/contract.rs

```diff
@@ -0,0 +1,340 @@
+use cosmwasm_std::{entry_point, Uint128};
+use cosmwasm_std::{Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use schemars::{JsonSchema, Map};
+use serde_derive::{Deserialize, Serialize};
+use serde_json_wasm::{from_str, to_string};
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{Authorization, Config, Kind, ALLOWED, CONFIG};
+use cw_auth_middleware::ContractError as AuthorizationError;
+
+const CONTRACT_NAME: &str = "crates.io:whitelist";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    U128(Uint128),
+    Number(u64), // Why doesn't u128 work here?
```

Comment:
u128 doesn't serialise directly thus Uint128, believe this is fixed in some version however

Reply:
I think I tried with `Uint128` but then *all* numbers became strings. Not sure if the solution I have now (numbers behave like numbers but large enough numbers require strings) is nice and user friendly or will just be confusing when it behaves differently for larger ints.